src/SeriesTracker.js

```diff
@@ -10,38 +10,77 @@ export function SeriesTracker(series) {
 
   this.add = function (serie) {
     this.series.push(serie);
-    //(If a serie has been watched) {
-    // Update the count of watched series: "numberOfWatched"
-    // Check for "lastSerie" property, set if we don't.
-    // Check for "lastSerie"'s finishedDate, if the serie's "finishedDate" prop is greater,
-    // set "lastSerie" prop to "serie" object.
-    //}
     if (serie.isWatched) {
+      //(If a serie has been watched) {
+      // Update the count of watched series: "numberOfWatched"
+      this.numberOfWatched += 1;
+      // Check for "lastSerie" property, set if we don't.
+      this.lastSerie ? {} : this.lastSerie = serie;
+      // Check for "lastSerie"'s finishedDate, if the serie's "finishedDate" prop is greater,
+      // set "lastSerie" prop to "serie" object.
+      const pattern = /(\d{2})\.(\d{2})\.(\d{4})/;
+      const parser = (dateString) => new Date(dateString.replace(pattern,'$3-$2-$1'));
+      parser(this.lastSerie.finishedDate) < parser(serie.finishedDate) ? this.lastSerie = serie : {};
+      
+    //}
+
     } else {
       // If a serie hasn't been watched:
       // Check if serie has "isCurrent" prop
-      // Check if we have a "currentSerie" property. Set if we don't.
-      // Check if we have a "nextSerie" property as well - if we didn't
-      // set the .currentSerie property, set the .nextSerie property.
-    }
-
-    //it should also update the number of series marked as watched / unwatched:
-    //"numberOfWatched" and "numberOfUnWatched"
+        // Check if we have a "currentSerie" property. Set if we don't.
+          if (serie.isCurrent && !this.currentSerie) {
+            this.currentSerie = serie;           
+          } 
+          
+          // Check if we have a "nextSerie" property as well - if we didn't
+          // set the .currentSerie property, set the .nextSerie property.
+          else if (!this.nextSerie) {
+            this.nextSerie = serie;
+          };   
+          //it should also update the number of series marked as watched / unwatched:
+          //"numberOfWatched" and "numberOfUnWatched"
+          this.numberOfUnWatched++;
+    };
+    
   };
 
   //check to see if we have series to process
   if (series.length > 0) {
     //Loop through all of the series in the "series" argument
+    for (let s of series) {
```

Comment:
for yerine forEach kullanılıarak, kod declarative, daha functional bir hale getirilebilir. 
```js
  if (series.length > 0) {
    series.forEach(this.add, this); // burda çağırılacak method this'e ihtiyaç duyduğu için ikinci parametrede yollamak zorundayız. Eğer direk bir add() methodu çağıracak olsaydık buna gerek kalmayacaktı. 
  }
```
Şu dökümanı inceleyebilirsin: https://betterprogramming.pub/functional-js-from-%CE%B1-to-%CF%89-8dc0cfe1f4e1